### ISSUE TITLE
Tasks: The security token included in the request is invalid

### ISSUE THREAD

**Issue by exp0nge** [Miserlou/Zappa#1043]
## Context
My AWS role has: `AdministratorAccess`. Running zappa in a VPC.

I've setup some tasks like this:
```
@task
def cache_tracks(tracks):
    if tracks:
        print("Caching tracks for {0}".format(tracks[0]['profile_id']))
    else:
        return

    for track in tracks:
        if track['origin'] == sources.SOURCE:
            # SOURCE
            requests.get('a/get/request/')
```

## Expected Behavior
The task should make some HTTP calls.

## Actual Behavior
Same behavior with sns
```Traceback (most recent call last):
  File "/var/task/django/core/handlers/exception.py", line 41, in inner
  response = get_response(request)
  File "/var/task/django/core/handlers/base.py", line 249, in _legacy_get_response
  response = self._get_response(request)
  File "/var/task/django/core/handlers/base.py", line 187, in _get_response
  response = self.process_exception_by_middleware(e, request)
  File "/var/task/django/core/handlers/base.py", line 185, in _get_response
  response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/var/task/django/views/generic/base.py", line 68, in view
  return self.dispatch(request, *args, **kwargs)
  File "/var/task/profiles/models.py", line 59, in dispatch
  return super().dispatch(request, *args, **kwargs)
  File "/var/task/django/views/generic/base.py", line 88, in dispatch
  return handler(request, *args, **kwargs)
  File "/var/task/songs/views.py", line 73, in get
  cache_tracks(songs)
  File "/var/task/zappa/async.py", line 341, in _run_async
  aws_region=aws_region).send(task_path, args, kwargs)
  File "/var/task/zappa/async.py", line 143, in send
  self._send(message)
  File "/var/task/zappa/async.py", line 157, in _send
  Payload=payload
  File "/var/task/botocore/client.py", line 253, in _api_call
  return self._make_api_call(operation_name, kwargs)
  File "/var/task/botocore/client.py", line 557, in _make_api_call
  raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (UnrecognizedClientException) when calling the Invoke operation: The security token included in the request is invalid
```

## Steps to Reproduce

1. Setup tasks
2. Call task
3.

## Your Environment

* Zappa version used: 0.43.1
* Operating System and Python version: Windows, 3.6
* The output of `pip freeze`:
* Link to your project (optional):
* Your `zappa_settings.py`:

**Comment by mjschultz** [comment]
As I mentioned in your #1049 PR, I think the problem is that you're _probably_ setting the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` values in one of the environments, but Lambda sets these by default along with `AWS_SESSION_TOKEN`. I suspect that the `AWS_SESSION_TOKEN` is what is causing the problem here.

You can probably look at [previous comment](https://github.com/Miserlou/Zappa/issues/984#issuecomment-313789220) explaining what I think is happening in more detail.

I think the two options you have would be to:
1. Simple give your Lambda role the access you need and remove your `AWS_*` environment variables from `zappa_settings` (or where ever they are).
2. Where you absolutely need to have your custom `AWS_*` access keys, instead of using a call like `boto3.client('service_name')` grab the custom keys from the environment and create the boto client that way (if applicable). It would look something like:
```
access_key_id = os.getenv('MY_ACCESS_KEY_ID')
secret_access_key = os.getenv('MY_SECRET_ACCESS_KEY')
client = boto3.client('service_name', aws_access_key_id=access_key_id, aws_secret_access_key=secret_access_key)
```

**Comment by exp0nge** [comment]
@mjschultz Thanks buddy. I think having django-storages is a common thing to have setup so I had those keys defined for that. I had then tested the async calls which fails. Changing the env names "solves" it.

I feel this should be noted somewhere.

**Comment by mjschultz** [comment]
I'm glad you solved the problem! I've previously filed #1000 to make it more obvious when something like this happens, but haven't written a solution because I don't think people would see CloudWatch debug messages easily.

That said I think you're right and a note somewhere in the docs might help a few people with this class of problem.
